### PR TITLE
Enable Azure Key Vault  for compute linked services

### PR DESCRIPTION
**Pull request by fhljys**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [Y] The title of the PR is clear and informative.
- [Y] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [Y] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [Y] If applicable, the PR references the bug/issue that it fixes.
- [Y] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [Y] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [Y] My spec meets the review criteria:
  - [Y] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [Y] The spec follows the guidelines described in the [Swagger checklist](../documentation/swagger-checklist.md) document.
  - [Y] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

File: specification/datafactory/resource-manager/Microsoft.DataFactory/preview/2017-09-01-preview/entityTypes/LinkedService.json [hunk 1]
```diff
@@ -237,7 +237,7 @@
         },
         "accessKey": {
           "description": "The Azure Batch account access key.",
-          "$ref": "../datafactory.json#/definitions/SecureString"
+          "$ref": "../datafactory.json#/definitions/SecretBase"
```
Comment (jianghaolu) [inline specification/datafactory/resource-manager/Microsoft.DataFactory/preview/2017-09-01-preview/entityTypes/LinkedService.json]: Cannot resolve reference. `"../../datafactory.json#/definitions/SecretBase"`?

**Review by jianghaolu — CHANGES_REQUESTED** [review]

**Review by jianghaolu — APPROVED** [review]

**Comment by jianghaolu** [comment]
Per communication offline, the error in semantic validation is a false alarm.

> Oops, this looks like a bug in travis-ci logic. So for PR_ONLY=true mode, we run the semantic validator on the files that were touched in this PR. Unfortunately this is the PR where only the file containing the model definitions (LinkedService.json) was touched. The actual swagger spec (datafactory.json) was not touched. The actual swagger spec should have been the entry point for the semantic validator instead of the external model file.
If I execute the following command on my machine, then everything works fine.
oav validate-spec https://github.com/fhljys/azure-rest-api-specs/blob/19db39b7ade68cf715a6d4102d6faf81d9215f30/specification/datafactory/resource-manager/Microsoft.DataFactory/preview/2017-09-01-preview/datafactory.json
info: No Errors were found.
Bottom line: in PR_ONLY true mode we need to ensure that we are validating a swagger spec and not a model definition file. The problem here is that model definition files are also swagger specs but they only have model definitions no paths. So we need to find an intelligent way to find out the right swagger spec(entry point) and validate accordingly.
Thanks,
Amar
